lib/IDE/SourceEntityWalker.cpp

```diff
@@ -608,16 +608,6 @@ ASTWalker::PreWalkResult<Expr *> SemaAnnotator::walkToExprPre(Expr *E) {
       return Action::Stop();
     // We already visited the children.
     return doSkipChildren();
-  } else if (auto ME = dyn_cast<MacroExpansionExpr>(E)) {
-    // Add a reference to the macro
-    auto macroRef = ME->getMacroRef();
-    if (auto *macroDecl = dyn_cast_or_null<MacroDecl>(macroRef.getDecl())) {
-      auto macroRefType = macroDecl->getDeclaredInterfaceType();
-      if (!passReference(
-              macroDecl, macroRefType, ME->getMacroNameLoc(),
-              ReferenceMetaData(SemaReferenceKind::DeclRef, llvm::None)))
-        return Action::Stop();
-    }
```

Comment:
Out of curiosity, why remove this? Wouldn't this impact things like cursor info?

Reply:
My assumption was that we end up passing the reference through some other means. But based on the failing test, seems like that isn't the case.

Reply:
OK, we needed to check that the `MacroExpansionExpr` didn’t have a `SubstituteDecl`. Could you double-check that the comment I wrote in my new version of this PR makes sense?

Reply:
Looks good to me!